hibernate-core/src/main/java/org/hibernate/query/internal/QueryInterpretationCacheStandardImpl.java

```diff
@@ -111,41 +111,49 @@ public HqlInterpretation resolveHqlInterpretation(
 		final StatisticsImplementor statistics = statisticsSupplier.get();
 		final boolean stats = statistics.isStatisticsEnabled();
 		final long startTime = ( stats ) ? System.nanoTime() : 0L;
-		final HqlInterpretation cached = hqlInterpretationCache.get( queryString );
-		if ( cached != null ) {
-			if ( stats ) {
-				statistics.queryPlanCacheHit( queryString );
-			}
-			return cached;
-		}
-
-		log.debugf( "Creating and caching HqlInterpretation - %s", queryString );
 
-		final SqmStatement<?> sqmStatement = creator.apply( queryString );
 		final DomainParameterXref domainParameterXref;
-		final ParameterMetadataImplementor parameterMetadata;
+		ImmutableHqlInterpretation immutableHqlInterpretation = hqlInterpretationCache.get( queryString );
+		if ( immutableHqlInterpretation == null ) {
+			log.debugf( "Creating and caching HqlInterpretation - %s", queryString );
+			final SqmStatement<?> sqmStatement = creator.apply( queryString );
+			final ParameterMetadataImplementor parameterMetadata;
+
+			if ( sqmStatement.getSqmParameters().isEmpty() ) {
+				domainParameterXref = DomainParameterXref.empty();
+				parameterMetadata = ParameterMetadataImpl.EMPTY;
+			}
+			else {
+				domainParameterXref = DomainParameterXref.from( sqmStatement );
+				parameterMetadata = new ParameterMetadataImpl( domainParameterXref.getQueryParameters() );
+			}
+
+			immutableHqlInterpretation = new ImmutableHqlInterpretation( sqmStatement, parameterMetadata);
```

Comment:
I think we can use 'final' above if only for consistency; otherwise, simply combine the two statements into one:
```
hqlInterpretationCache.put( queryString, new ImmutableHqlInterpretation( sqmStatement, parameterMetadata ) );
```
We might well insert a space char before the ending ')' above as well, though trivial.